### PR TITLE
[PLATFORM-1241] Rename community product to data union

### PR DESCRIPTION
**Pull request by juhah**
Renamed instances and code that reference community product to data union, with these exceptions:

- ~Left the `process.env.COMMUNITY_PRODUCTS` flag as it was since it's a temporary solution~
- Added a new `DATA_UNION` flag to `productTypes` in `$shared/utils/constanct`, but the old `COMMUNITY` is used to create a product until the API changes
- Community product smart contract and ABIs are still called that so left as it is until/if they are updated
- Api endpoint paths not changed until they are renamed

Flag names have been changed and need to be redefined in `.env`:
`COMMUNITY_PRODUCTS=on` -> `DATA_UNIONS=on`
`COMMUNITY_PRODUCT_OPERATOR_ADDRESS=...` -> `DATA_UNION_OPERATOR_ADDRESS=...`
`COMMUNITY_PRODUCT_BLOCK_FREEZE_PERIOD_SECONDS=...` -> `DATA_UNION_BLOCK_FREEZE_PERIOD_SECONDS=...`

Lots of file changes so no point to go though all the code but a have look in the UI and see that everything still works.

### CODE REVIEW

File: app/src/marketplace/modules/dataUnion/services.js [hunk 1]
```diff
@@ -189,75 +189,92 @@ export const getCommunityData = async (id: CommunityId, usePublicNode: boolean =
     }
 }
 
-export const getSecrets = (communityId: CommunityId): ApiResult<Array<Secret>> =>
-    get({
-        url: formatApiUrl('communities', communityId, 'secrets'),
-    })
+type GetSecrects = {
```
Comment (mondoreale) [inline app/src/marketplace/modules/dataUnion/services.js]: Typo?

**Comment by fonty1** [comment]
I don't think there's any harm renaming the contracts in `src/shared/web3/config.js` but up to you.

Some missed CP mentions:
- `CP_SERVER_POLL_INTERVAL_MS`
- `cpStats`
- `skipCpDeployGuide`
- `// allow 5s for the API to start in CP server`

Are these env vars temporary also? 
- `COMMUNITY_PRODUCT_OPERATOR_ADDRESS`
- `COMMUNITY_PRODUCT_BLOCK_FREEZE_PERIOD_SECONDS`

**Comment by juhah** [comment]
@fonty1 
> Some missed CP mentions:

Thanks, I will change them 👍 

> Are these env vars temporary also?
> COMMUNITY_PRODUCT_OPERATOR_ADDRESS
> COMMUNITY_PRODUCT_BLOCK_FREEZE_PERIOD_SECONDS

No, these ones are not. I was hoping to get away without changing `.env` variables to avoid problems that come when you forget to define them once this gets merged. But i guess it's ok

**Comment by juhah** [comment]
Changed the flag names, you'll need to define these in `.env`:
`DATA_UNIONS=on`
`DATA_UNION_OPERATOR_ADDRESS=...`
`DATA_UNION_BLOCK_FREEZE_PERIOD_SECONDS=...`

**Pull request merged**